src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs

```diff
@@ -296,9 +315,30 @@ private void MaybeSetCache(GetResponse getResponse, GetRequest command)
             if (changeVector == null)
                 return;
 
-            _cache.Set(cacheKey, changeVector, result);
+            _httpCache.Set(cacheKey, changeVector, result);
         }
 
         public override bool IsReadRequest => false;
+
+        public void Dispose() => _cached?.Dispose();
+
+        private class Cached : IDisposable
+        {
+            public readonly (HttpCache.ReleaseCacheItem Release, BlittableJsonReaderObject Cached)[] Values;
+
+            public Cached(int size)
+            {
+                Values = ArrayPool<(HttpCache.ReleaseCacheItem, BlittableJsonReaderObject)>.Shared.Rent(size);
+            }
+
+            public void Dispose()
```

Comment:
It does not check if it was disposed before. Will it result in double disposal since we are disposing it earlier as well?

Reply:
I set the Cache to null after disposing but I will do additional check here